packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts

```diff
@@ -126,8 +126,10 @@ export const useGridKeyboardNavigation = (
         }
 
         case 'PageUp': {
-          if (rowIndexBefore - viewportPageSize >= firstRowIndexInPage) {
-            goToCell(colIndexBefore, rowIndexBefore - viewportPageSize);
+          // Go to the first row before going to header
+          const nextRowIndex = Math.max(rowIndexBefore - viewportPageSize, 0);
+          if (nextRowIndex !== rowIndexBefore && nextRowIndex >= firstRowIndexInPage) {
```

Comment:
The behavior should be the same on the page > 0, right now only the 1st page has your fix.

+ add a test on page 1 :+1:

Reply:
Wow, nice catch!
Indeed, I didn't think about pagination,, thanks!